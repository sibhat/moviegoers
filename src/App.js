// import React, { Component } from "react";
// // import axios from "axios";
// import "./App.css";
// import DetailContainer from "./feature/detail/container";
import ListContainer from "./feature/list/container";
import FooterContainer from "./feature/footer/container";
import Main from "./feature/main/container";
// import Nav from "./feature/nav/presentational";
// import { Route } from "react-router-dom";
import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { styles } from "./AppJSS";
import { config } from "./feature/list/store/action";
import { connect } from "react-redux";
class App extends React.Component {
	state = {
		open: false
	};
	componentDidMount() {
		this.props.config();
	}

	handleDrawerOpen = () => {
		this.setState({ open: true });
	};

	handleDrawerClose = () => {
		this.setState({ open: false });
	};

	render() {
		const { classes } = this.props;

		return (
			<div className={classes.root}>
				<CssBaseline />
				{/* <Nav
					classes={classes}
					open={this.state.open}
					handleDrawerOpen={this.handleDrawerOpen}
					handleDrawerClose={this.handleDrawerClose}
				/> */}

				<main className={classes.content}>
					<Main className={classes.main} />
					<ListContainer status="now_playing" />

					<ListContainer status="top_rated" />
					{/* <ListContainer status="latest" /> */}
					<ListContainer status="popular" />
					<ListContainer status="upcoming" />

					{/* <Route path="/movies/:id" component={DetailContainer} /> */}
				</main>
				<FooterContainer />
			</div>
		);
	}
}

App.propTypes = {
	classes: PropTypes.object.isRequired
};

export default withStyles(styles)(
	connect(
		null,
		{ config }
	)(App)
);
