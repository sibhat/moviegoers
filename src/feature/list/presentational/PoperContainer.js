import React from "react";
import {
	Popper,
	Fade,
	Paper,
	Typography,
	Divider,
	Button
} from "@material-ui/core";

import { Card, Dialog, DialogActions, DialogContent } from "@material-ui/core";

import { withRouter, Link } from "react-router-dom";
import "./hover.css";

import Main from "../../main/container";

let x = 0;
let timeInterval;
const PoperContainer = props => {
	let { movie, styles, classes, history } = props;
	const [anchorEl, setAnchorEl] = React.useState(null);
	const [open, setOpen] = React.useState(true);
	const [dOpen, setDOpen] = React.useState(false);
	function moreInfoHandler(e) {
		let option = movie.title ? "movies" : "tv";
		history.push(`/${option}/${movie.id}`);
	}
	function handlerFunc(e) {
		// const { currentTarget } = e;
		// setAnchorEl(currentTarget);
		setDOpen(!dOpen);

		props.movieDetailHandler(movie.id);
	}
	function onMouseEnterHandler(e) {
		const { currentTarget } = e;
		x = 0;
		timeInterval = setInterval(() => {
			x++;
			if (x > 3) {
				x = 0;
				setAnchorEl(currentTarget);
				setOpen(true);
			}
		}, 100);
	}
	function onMouseLeaveHandler() {
		setAnchorEl(null);
		setOpen(false);
		clearInterval(timeInterval);
	}

	return (
		<>
			<div
				className={styles.card + " card"}
				key={movie.id}
				onClick={handlerFunc}
			>
				<img
					className={styles.img}
					alt="poster"
					src={`${props.baseUrl}/${props.size}${movie.poster_path}`}
				/>

				{/* {({ TransitionProps }) => (
						<Fade {...TransitionProps} timeout={350}> */}
				<Paper className={classes.popper + " popper"} sibhat={"true"}>
					<img
						className={classes.bgCard}
						alt="poster"
						src={"../assets/04.png"}
					/>
					<span className={classes.bgCardSpan} />
					<Typography className={classes.title}>
						{movie.title || movie.name}
					</Typography>
					<div className={classes.flex}>
						<Typography className={classes.description}>
							{(movie.release_date &&
								movie.release_date.slice(0, 4)) ||
								(movie.first_air_date &&
									movie.first_air_date.slice(0, 4))}
						</Typography>
						<Typography className={classes.description}>
							{movie.vote_average || movie.popularity}
						</Typography>
					</div>
					<Divider />
					<Typography className={classes.description}>
						{movie.overview}
					</Typography>
					<Divider />
					<Divider />
				</Paper>
				{/* </Fade> */}
				{/* )} */}
			</div>

			<Dialog
				className={classes.dialog}
				fullWidth={true}
				open={dOpen}
				onClose={handlerFunc}
			>
				<div className={classes.flex}>
					<DialogContent className={classes.main}>
						<Main />
					</DialogContent>
					<div className={classes.dialogSection}>
						<Typography className={classes.title}>
							{movie.title || movie.name}
						</Typography>
						<div className={classes.flex}>
							<Typography className={classes.description}>
								{(movie.release_date &&
									movie.release_date.slice(0, 4)) ||
									(movie.first_air_date &&
										movie.first_air_date.slice(0, 4))}
							</Typography>
							<Typography className={classes.description}>
								{movie.vote_average || movie.popularity}
							</Typography>
						</div>
						<Divider />
						<Typography className={classes.description}>
							{movie.overview}
						</Typography>
					</div>
				</div>
				<DialogActions>
					<Button onClick={handlerFunc} color="primary" autoFocus>
						Close
					</Button>
					<Button onClick={moreInfoHandler} color="primary" autoFocus>
						more info
					</Button>
				</DialogActions>
			</Dialog>
		</>
	);
};

export default withRouter(PoperContainer);
