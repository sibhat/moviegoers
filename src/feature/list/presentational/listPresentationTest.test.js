import React from "react";
import { shallow, mount } from "enzyme";
import List from "./index";
import PoperContainer from "./PoperContainer";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "../../../reducer";
import App from "../../../App";

const store = configureStore();

describe("<List />", () => {
	it("renders component without problem", () => {
		const wrapper = shallow(<List />);

		expect(wrapper).toMatchSnapshot();
	});
	it("renders card component without problem", () => {
		let data = {
			results: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }]
		};
		const listWrapper = shallow(<List data={data} />);
		const poperWrapper = listWrapper.find(".card");

		expect(poperWrapper.exists).toBeTruthy();
	});
});
describe("<PoperContainer />", () => {
	it("renders component without problem", () => {
		const cardWrapper = shallow(<PoperContainer />);

		expect(cardWrapper).toMatchSnapshot();
	});
});
