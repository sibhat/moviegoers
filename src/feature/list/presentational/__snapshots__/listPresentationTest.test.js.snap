// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders component without problem 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(List) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "bgCard": "List-bgCard-6",
        "bgCardSpan": "List-bgCardSpan-7",
        "card": "List-card-3",
        "catagory": "List-catagory-2",
        "description": "List-description-9",
        "dialog": "List-dialog-11",
        "dialogSection": "List-dialogSection-13",
        "flex": "List-flex-10",
        "img": "List-img-5",
        "main": "List-main-12",
        "popper": "List-popper-4",
        "root": "List-root-1",
        "title": "List-title-8",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "bgCard": "List-bgCard-6",
          "bgCardSpan": "List-bgCardSpan-7",
          "card": "List-card-3",
          "catagory": "List-catagory-2",
          "description": "List-description-9",
          "dialog": "List-dialog-11",
          "dialogSection": "List-dialogSection-13",
          "flex": "List-flex-10",
          "img": "List-img-5",
          "main": "List-main-12",
          "popper": "List-popper-4",
          "root": "List-root-1",
          "title": "List-title-8",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PoperContainer /> renders component without problem 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(PoperContainer) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
